pkg/instrumentation/sdk.go

```diff
@@ -432,8 +449,12 @@ func chooseServiceVersion(pod corev1.Pod, index int) string {
 
 // creates the service.instance.id following the semantic defined by
 // https://github.com/open-telemetry/semantic-conventions/pull/312.
-func createServiceInstanceId(namespaceName, podName, containerName string) string {
-	var serviceInstanceId string
+func createServiceInstanceId(pod corev1.Pod, namespaceName, podName, containerName string) string {
+	serviceInstanceId := chooseLabelOrAnnotation(pod, semconv.ServiceInstanceIDKey, constants.AnnotationAppInstance)
+	if serviceInstanceId != "" {
+		return serviceInstanceId
+	}
```

Comment:
I'm concerned that this would be a breaking change.

Reply:
fixed